Documentation/Profiling/IL Rewriting Basics.md

```diff
@@ -0,0 +1,23 @@
+# IL Rewriting Basics
+
+## Intro
+One of the common use cases of the `ICorProfiler*` interfaces is to perform IL rewriting. Some possible reasons a profiler would want to rewrite IL:
+- Inspecting interesting process state
+  - Capturing exception state
+  - Inspecting managed objects
+  - Inspecting function arguments/return values
+- Injecting method hooks at the start/end of the method that call in to another managed library
+
+There are two ways to rewrite IL
+
+1. At Module load time with `ICorProfilerInfo::SetILFunctionBody`
+    This approach has the benefit that it is 'set it and forget it'. You can replace the IL at module load, and the runtime will treat this new IL as if the module contained that IL - you don't have to worry about any of the quirks of ReJIT. The downside is that is is unrevertable - once it is set, you cannot change your mind.
+
+2. At any point during the process lifetime with 'ICorProfilerInfo4::RequestReJIT' or 'ICorProfilerInfo10::RequestReJITWithInliners'.
```

Comment:
If you want to be consistent with the other places in the doc, these need to be surrounded with backticks, i.e:

\`ICorProfilerInfo4::RequestReJIT\` or \`ICorProfilerInfo10::RequestReJITWithInliners\`

Reply:
Thanks!